zcash_history/src/node_data.rs

```diff
@@ -13,10 +14,13 @@ pub const MAX_NODE_DATA_SIZE: usize = 32 + // subtree commitment
     32 + // subtree total work
     9 +  // start height (compact uint)
     9 +  // end height (compact uint)
-    9; // Sapling tx count (compact uint)
-       // = total of 171
+    9 + // Sapling tx count (compact uint)
+    32 + // start Orchard tree root
+    32 + // end Orchard tree root
+    9; // Orchard tx count (compact uint)
+       // = total of 244
```

Comment:
In Zebra we're using `MAX_NODE_DATA_SIZE` to create fixed-size byte arrays.

So between merging this PR and NU5 activation, we would have some unused trailing zero bytes. But the RAM and disk impact is minimal because we only store one copy of the cache. So we just need to check that we're not expecting to read or write the whole array every time.

@conradoplg does https://github.com/ZcashFoundation/zebra/pull/2227 assume that we're reading or writing every node data byte?

Reply:
This constant is intentionally the upper bound on the nose data encoding, to reflect the fact that some of the data types are compact sizes, and thus variable width. This size is used for e.g. predictable array sizes crossing the FFI in `zcashd`.

Reply:
ZcashFoundation/zebra#2227 just hands it over to librustzcash (`zcash_history::NodeData::from_bytes`) and doesn't assume anything else